src/twisted/words/test/test_irc.py

```diff
@@ -2572,6 +2572,71 @@ def test_dccChatIndecipherablePort(self):
         self.assertEqual(str(result), "Indecipherable port 'sd@d'")
 
 
+UNICODE_REPLACEMENT_CHARACTER = "\ufffd"
+
+
+class ClientDecodingTests(IRCTestCase):
+    """
+    Tests for L{IRCClient} receiving data in different encodings.
+    """
+
+    def setUp(self):
+        methods = ["handleCommand", "badMessage"]
+        self.client = CollectorClient(methods)
+
+    def test_decoding_unspecified_codec(self):
```

Comment:
Thank you for writing such nice docstrings!

The test method names (and locals) should also be camelCase. Really minor nit: you can [remove "Test that" from the docstring](https://jml.io/pages/test-docstrings.html) as it's redundant.